File: src/main/scala/io/github/hexirp/akariki/Context.java
package io.github.hexirp.akariki;

import net.minecraft.util.ResourceLocation;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * {@link Context} is data of akariki.
 */
public class Context {
    private final String id;
    private final Logger log;

    public Context(String id) {
        this.id = id;
        this.log = LogManager.getFormatterLogger(id);
    }

    public ResourceLocation newResourceLocation(String name) {
        return new ResourceLocation(id, name);
    }

    public void info(String message) {
        log.info(message);
    }
}

File: src/main/scala/io/github/hexirp/akariki/Context.scala
package io.github.hexirp.akariki

import net.minecraft.util.ResourceLocation

import org.apache.logging.log4j.{LogManager, Logger}

/** [[Context]] is data of akariki.
  *
  * @param id The ID of a MOD
  */
class Context(id : String) {
  val log : Logger = LogManager.getFormatterLogger(id)

  def newResourceLocation(name : String) : ResourceLocation = new ResourceLocation(id, name)

  def info(message : String) : Unit = LogManager.getFormatterLogger(id).info(message)
}
